chronos/internal/asyncfutures.nim

```diff
@@ -52,6 +54,78 @@ type
     ## Future to hold GC seqs
     gcholder*: seq[B]
 
+  SomeFuture = Future|InternalRaisesFuture
+
+macro prepend(tup: typedesc[tuple], typ: varargs[typed]): typedesc =
```

Comment:
Is it possible to move this code to `asyncmacro.nim`? I really do not want to see macro code here.

Reply:
> Is it possible to move this code to asyncmacro.nim? 

messy - these should not become public, at least not with these names - they're not meant to be part of the API (yet) - later, maybe it makes sense to expose exception-list-manipulation-helpers but that would be a new module I think - they're here for convenience mainly.

Reply:
moved to its own module - 03dd6ed104755a67fda2c5c733e84dd139ab4ec1